help.py

```diff
@@ -41,3 +42,276 @@ def datetime_to_epoch_millis(date: datetime) -> int:
+
+        if args.days and args.minutes:
+            sys.exit('--days and --minutes are mutually exclusive')
+
+        if (args.sigmarule or args.sigmadir) and edr == 'cortex':
+            sys.exit('Neither --sigmarule nor --sigmadir are supported by product "cortex"')
+
+        if (args.sigmarule or args.sigmadir) and edr == 's1' and not args.dv:
+            sys.exit('Neither --sigmarule nor --sigmadir are supported by SentinelOne PowerQuery')
+
+        if args.sigmarule and not os.path.isfile(args.sigmarule):
+            sys.exit('Supplied --sigmarule is not a file')
+
+        if args.sigmadir and not os.path.isdir(args.sigmadir):
+            sys.exit('Supplied --sigmadir is not a directory')
+
+        log = logger(edr=args.edr,logs_dir=args.log_dir)
+
+        definition_files = list()
+        definitions = dict()
+        sigma_rules = list()
+
+        # test if deffile exists
+        # deffile can be resolved from 'definitions' folder without needing to specify path or extension
+        if args.deffile:
+            if not os.path.exists(args.deffile):
+                repo_deffile: str = os.path.join(os.path.dirname(__file__), 'definitions', args.deffile)
+                if not repo_deffile.endswith('.json'):
+                    repo_deffile = repo_deffile + '.json'
+
+                if os.path.isfile(repo_deffile):
+                    log.debug(f'Using repo definition file {repo_deffile}')
+                    args.deffile = repo_deffile
+                else:
+                    sys.exit("The deffile doesn't exist. Please try again.")
+            definition_files.append(args.deffile)
+
+        # if --defdir add all files to list
+        if args.defdir:
+            if not os.path.exists(args.defdir):
+                sys.exit("The defdir doesn't exist. Please try again.")
+            else:
+                for root_dir, dirs, files in os.walk(args.defdir):
+                    for filename in files:
+                        if os.path.splitext(filename)[1] == '.json':
+                            definition_files.append(os.path.join(root_dir, filename))
+
+        if definitions: survey_payload['definitions'] = definitions
+
+        # add sigmarule to list
+        if args.sigmarule:
+            sigma_rules.append(args.sigmarule)
+
+        # if --sigmadir, add all files to sigma_rules list
+        if args.sigmadir:
+            for root_dir, dirs, files in os.walk(args.sigmadir):
+                for filename in files:
+                    if os.path.splitext(filename)[1] == '.yml':
+                        sigma_rules.append(os.path.join(root_dir, filename))
+
+        if sigma_rules: survey_payload['sigma_rules'] = sigma_rules
+
+        # run search based on IOC file
+        if args.iocfile:
+            with open(args.iocfile) as ioc_file:
+                basename = os.path.basename(args.iocfile)
+                ioc_source = basename
+                ioc_list = [x.strip() for x in ioc_list]
```

Comment:
Should this be `ioc_file` instead?

```suggestion
                ioc_list = [x.strip() for x in ioc_file]
```